csrc/preseg_passes/allocation_order_inference.cpp

```diff
@@ -240,6 +323,12 @@ void AllocationOrderInferencer::handle(BroadcastOp* op) {
     return;
   }
 
+  // propagate empty allocation order;
+  if (iter->second.empty()) {
+    alloc_order_map_[out] = {};
+    return;
+  }
```

Comment:
More generally, if all inputs have empty allocation orders, we should propagate as empty, right? If so, then maybe this could be done in `dispatch(Expr*)` instead of handling for each expr type.

Reply:
That's a good point. So `dispatch(Expr*)` should hijack the dispatch.

I'll start another PR playing with that. :+1: